src/su.c

```diff
@@ -655,8 +655,7 @@ static /*@only@*/struct passwd * check_perms (void)
 		SYSLOG ((LOG_INFO,
 		         "Change user from '%s' to '%s' as requested by PAM",
 		         name, tmp_name));
-		strncpy (name, tmp_name, sizeof(name) - 1);
```

Comment:
So you're right in the patch description that

> tmp_name' has been passed to strcmp(3) a few lines above.

however that could have been counting on $name being null terminated, not $tmp_name.

Now I find it hard to believe that pam_get_item would return a non-null-terminated string with PAM_SUCCESS, however I :sadface: notice that the manpage declines to guarantee that for us :(

Reply:
Taking into account that there's no size parameter, it necessarily is a string.  Otherwise, it would need to document some kind of size that the user needs to use to read from the pointer, to avoid overrunning the buffer.

I guess they didn't document it believing it would be obvious...  I'd prefer it documented though.  Just saying that it's a string would be enough :/

Reply:
BTW, `strcmp(3)` requires that both arguments are strings.  If one of them is not a string, then `strncmp(3)` should be called.